packages/components/src/custom-select-control/test/index.js

```diff
@@ -238,28 +248,56 @@ describe.each( [
 		).toHaveTextContent( 'Hint' );
 	} );
 
-	it( 'shows selected hint in list of options when added, regardless of __experimentalShowSelectedHint prop', async () => {
+	it( 'Should accept and pass arbitrary properties to the selectedItem object in the onChange callback', async () => {
 		const user = userEvent.setup();
+		const onChangeMock = jest.fn();
 
 		render(
 			<Component
 				{ ...props }
-				label="Custom select"
 				options={ [
+					...props.options,
 					{
-						key: 'one',
-						name: 'One',
-						__experimentalHint: 'Hint',
+						key: 'custom',
+						name: 'Custom Option',
+						className: 'custom-class-name',
+						customProp1: 'value1',
+						customProp2: 42,
+						style: {
+							backgroundColor: 'rgb(127, 255, 212)',
+							rotate: '13deg',
+						},
 					},
 				] }
+				onChange={ onChangeMock }
 			/>
 		);
 
+		const currentSelectedItem = screen.getByRole( 'button', {
+			expanded: false,
+		} );
+
+		await user.click( currentSelectedItem );
 		await user.click(
-			screen.getByRole( 'button', { name: 'Custom select' } )
+			screen.getByRole( 'option', { name: 'Custom Option' } )
 		);
 
-		expect( screen.getByRole( 'option', { name: /hint/i } ) ).toBeVisible();
+		expect( onChangeMock ).toHaveBeenCalledTimes( 1 );
```

Comment:
We should add the same assertions added for v2 checking that `customProp1` and `customProp2` are not added as props/attributes of the item

Reply:
Added same test to v1